covidtask/covidtask/views.py

```diff
@@ -1,53 +1,82 @@
 from rest_framework.response import Response
-from rest_framework.decorators import api_view,permission_classes,action
-from .models import Country
-from .serializers import CountrySerializer,SubscribtionSerializer
-from rest_framework.generics import ListAPIView,CreateAPIView
-from rest_framework.viewsets import ViewSetMixin,ViewSet
-from rest_framework import views,status
+from .models import Country, CountryDailyCases
+from .serializers import CountrySerializer, SubscribtionSerializer, CountryViewSerializer
+from rest_framework.generics import ListAPIView, CreateAPIView
+from rest_framework import views, status
 from django.db.models import Count
-from django.contrib.auth.models import User
 
-class CountryViewSet(ViewSetMixin,ListAPIView):
+
+class CountryPercentageView(ListAPIView):
     serializer_class = CountrySerializer
     queryset = Country.objects.all()
-    @action(detail=False,methods=('GET',),url_path="death-percentage")
-    def get_percentage(self,request,*args,**kwargs):
-        if 'country' not in request.query_params:
-            return Response({'message': 'Missing parameters (country)', 'error': True, 'code': 400}, status=status.HTTP_400_BAD_REQUEST)
+
+    def get(self, request, country, *args, **kwargs):
+
+        country = Country.objects.filter(name=country).first()
+        if not country:
+            return Response({'message': 'Country not found', 'error': False, 'code': 400},
+                            status=status.HTTP_400_BAD_REQUEST)
         try:
-            country = Country.objects.filter(name=request.query_params['country']).first()
-            countrySerializer = CountrySerializer(country)
-            deathCount = countrySerializer.data['death']
-            confirmedCount = countrySerializer.data['confirmed']
-            percentage = deathCount / confirmedCount
-            responseData = {"death": deathCount, "confirmed": confirmedCount, "percentage": percentage}
-            return Response(responseData)
-        except:
-            return Response()
-
-    @action(detail=False, methods=('GET',), url_path="top")
-    def get(self, request, *args, **kwargs):
-        if 'status' not in request.query_params:
-            return Response({'message': 'Missing parameters (status)', 'error': True, 'code': 400}, status=status.HTTP_400_BAD_REQUEST)
-        stauts = request.query_params['status']
-        if 'limit' in request.query_params:
-            limit = int(request.query_params['limit'])
+            country_history = country.country_daily_cases.latest('created_at')
```

Comment:
Please get the totals using Django aggregation querysets instead of storing the totals.